### PR TITLE
fix admin browser history duplications

### PR DESCRIPTION
**Pull request by viniagostini**
Workspace with the linked solution: https://agostini--storecomponents.myvtex.com/admin/

This solves the same problem that the #3 but without breaking the SPAs that run on admin iframe (hopefully 🙏 )

## Description:  
The admin history was getting two entries when navigating from one page to another due to iframe's src URL updating. To fix that we now reload the entire iframe for each navigation.

## Before: 
history_before

## After
history_after

### CODE REVIEW

File: react/components/Iframe.js [hunk 1]
```diff
@@ -87,27 +87,27 @@ class Iframe extends Component {
...
   updateBrowserHistory = ({ pathname: iframePathname, search: iframeSearch, hash: iframeHash }) => {
     const { isDeloreanAdmin } = this.props
-    const { navigate } = this.context
     const { pathname, search = '', hash = '' } = window.location
     const patchedIframeSearch = iframeSearch.replace(/(\?|\&)env\=beta/, '')
 
-    if (iframePathname.replace(isDeloreanAdmin ? '/iframe' : '/app', '') !== pathname
+    let formatedIframePathname = iframePathname.replace(isDeloreanAdmin ? '/iframe' : '/app', '')
+
+    const formatedPathname = !pathname.endsWith('/') ? `${pathname}/` : pathname
```
Comment (vibegui) [inline react/components/Iframe.js:95]: nit: could be a function `withTrailingSlash()`

Comment (vibegui) [reply react/components/Iframe.js:95]: Also, wouldn't it be better to _remove_ the trailing slash by default? Apparently this thingy here which is used by most admins only adds trailing slash if there is a hash: https://io.vtex.com.br/oms-ui/2.31.1/lib/vtex-verify-url-slash.js

File: react/components/Iframe.js [hunk 2]
```diff
@@ -119,32 +119,33 @@ class Iframe extends Component {
...
           : search + '&env=beta'
-    const nextSrc = this.buildSrc(nextProps, patchedSearch, hash)
+    return `/admin/${props.isDeloreanAdmin ? 'iframe' : 'app'}/${props.params.slug || ''}${patchedSearch || ''}${hash || ''}`
+  }
 
-    return nextProps.isDeloreanAdmin || !(this.iframe && this.iframe.contentWindow.location.pathname.startsWith(nextSrc))
+  componentDidUpdate() {
+    const nextSrc = this.buildSrc(this.props)
+    if (this.props.isDeloreanAdmin || !(this.iframe && this.iframe.contentWindow.location.pathname === nextSrc)){
+      this.iframe && this.iframe.contentWindow.location.replace(nextSrc)
```
Comment (vibegui) [inline react/components/Iframe.js:128]: Would be nice to have a comment on why this happens

File: react/components/Iframe.js [hunk 3]
```diff
@@ -119,32 +119,33 @@ class Iframe extends Component {
...
-          : search + '&env=beta'
-
-    const src = location.pathname.replace('/admin', isDeloreanAdmin ? '/admin/iframe' : '/admin/app') + patchedSearch + hash
 
+    // if we update the iframe src, the iframe redirects to the src address and it automatically propagates this navigation to
+    // the browser history (this is the expected behaviour https://trillworks.com/nick/2014/06/11/changing-the-src-attribute-of-an-iframe-modifies-the-history/)
+    // so we only want to set the iframe src when it has been created in order to start on the URL that has been inserted in the browser in the first navigation.
+    // the navigation from outside into the iframe is made by `componentDidUpdate` changing the window location, this doesnt affect the browser history.
+    if (this.fixedSrc === '') {
+      this.fixedSrc = this.buildSrc(this.props)
```
Comment (vibegui) [inline react/components/Iframe.js:146]: Changing to different admins has no consequence? Nice

Comment (vibegui) [reply react/components/Iframe.js:146]: I would probably prefer something like returning false on `shouldComponentUpdate` if the iframe has been mounted, but this is not a major issue. Go ahead, merge this and let's test it out.

**Review by vibegui — APPROVED** [review]
Nice! 🌟

**Comment by viniagostini** [comment]
Nice! Thanks for the comments, i will make these changes in another PR

**Pull request merged**